src/components/inputs/accent-range.tsx

```diff
@@ -1,28 +1,75 @@
-import React from 'react';
+import React, {useState, useRef, useEffect} from 'react';
 import styled from 'styled-components';
 
 const Container = styled.span`
-  display: inline-block;
+  display: flex;
+  align-items: center;
   line-height: initial;
-  width: 200px;
+  width: 250px; // Increase the width
 `;
 
-const SliderInput = styled.input.attrs({type: 'range'})<any>`
+const SliderInput = styled.input.attrs({type: 'range'})`
   accent-color: var(--color_accent);
   width: 100%;
 `;
 
-export const AccentRange: React.FC<
-  Omit<React.InputHTMLAttributes<HTMLInputElement>, 'onChange'> & {
-    onChange: (x: number) => void;
+const ValueDisplay = styled.span`
+  margin-right: 10px;
+  width: 50px; // Set a fixed width
+  text-align: right; // Align the text to the right
+  flex-shrink: 0; // Prevent it from shrinking
+`;
+
+interface AccentRangeProps
+  extends Omit<
+    React.InputHTMLAttributes<HTMLInputElement>,
+    'onChange' | 'type'
+  > {
+  onChange: (x: number) => void;
+  showingSliderValue?: boolean; // Make showSliderValue optional
+}
+
+export const AccentRange: React.FC<AccentRangeProps> = (props) => {
+  const {showingSliderValue = false} = props; // Default showSliderValue to false if not provided
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [sliderValue, setSliderValue] = useState<number>(0); // Initialize with 0
+
+  useEffect(() => {
+    if (inputRef.current) {
+      setSliderValue(+inputRef.current.value || 0); // Provide a default value
+    }
+  }, [props.value]);
+
+  if (!showingSliderValue) {
+    // Render original version without value display
+    return (
+      <Container>
+        <SliderInput
+          ref={inputRef}
+          {...props}
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+            const value = +e.target.value;
+            setSliderValue(value);
+            props.onChange?.(value);
+          }}
+        />
+      </Container>
+    );
   }
-> = (props) => (
-  <Container>
-    <SliderInput
-      {...props}
-      onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
-        props.onChange && props.onChange(+e.target.value);
-      }}
-    />
-  </Container>
-);
+
+  // Render enhanced version with value display
+  return (
+    <Container>
+      <ValueDisplay>{sliderValue}</ValueDisplay>
+      <SliderInput
+        ref={inputRef}
+        {...props}
+        onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+          const value = +e.target.value;
+          setSliderValue(value);
+          props.onChange?.(value);
+        }}
+      />
+    </Container>
+  );
+};
```

Comment:
so u dont need the if else just for `showingSliderValue`
```suggestion
  return (
    <Container>
    { showingSliderValue && 
    	<ValueDisplay>{sliderValue}</ValueDisplay>
    }
      <SliderInput
        ref={inputRef}
        {...props}
        onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
          const value = +e.target.value;
          setSliderValue(value);
          props.onChange?.(value);
        }}
      />
    </Container>
  );
};
```